lucet-module/src/module_data.rs

```diff
@@ -34,6 +34,38 @@ pub struct ModuleData<'a> {
     export_functions: Vec<ExportFunction<'a>>,
     signatures: Vec<Signature>,
     module_signature: Vec<u8>,
+    features: ModuleFeatures,
+}
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct ModuleFeatures {
+    pub sse3: bool,
+    pub ssse3: bool,
+    pub sse41: bool,
+    pub sse42: bool,
+    pub avx: bool,
+    pub bmi1: bool,
+    pub bmi2: bool,
+    pub lzcnt: bool,
+    pub popcnt: bool,
+    _hidden: (),
```

Comment:
What's this for?

Reply:
It's to prevent exhaustive matching on the struct by naming all fields. Since all the other fields of this struct are pub, you _could_ exhaustively match that way, meaning adding fields is a semver-breaking change. I'm particularly uncomfortable with that because this is in `lucet-module` which my mental model pairs with a pretty vigilant stance around versioning.